pkg/chaosctl/cmd/debug.go

```diff
@@ -0,0 +1,200 @@
+			if err := o.Run("networkchaos", args, c); err != nil {
+				log.Fatal(err)
+			}
+		},
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			if len(args) != 0 {
+				return nil, cobra.ShellCompDirectiveNoFileComp
+			}
+			return listChaos("networkchaos", o.namespace, toComplete, c.CtrlCli)
+		},
+	}
+
+	stressCmd := &cobra.Command{
+		Use:   `stresschaos (CHAOSNAME) [-n NAMESPACE]`,
+		Short: `Print the debug information for certain stress chaos`,
+		Long:  `Print the debug information for certain stress chaos`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := o.Run("stresschaos", args, c); err != nil {
+				log.Fatal(err)
+			}
+		},
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			if len(args) != 0 {
+				return nil, cobra.ShellCompDirectiveNoFileComp
+			}
+			return listChaos("stresschaos", o.namespace, toComplete, c.CtrlCli)
+		},
+	}
+
+	ioCmd := &cobra.Command{
+		Use:   `iochaos (CHAOSNAME) [-n NAMESPACE]`,
+		Short: `Print the debug information for certain io chaos`,
+		Long:  `Print the debug information for certain io chaos`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := o.Run("iochaos", args, c); err != nil {
+				log.Fatal(err)
+			}
+		},
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			if len(args) != 0 {
+				return nil, cobra.ShellCompDirectiveNoFileComp
+			}
+			return listChaos("iochaos", o.namespace, toComplete, c.CtrlCli)
+		},
+	}
+
+	debugCmd.AddCommand(networkCmd)
+	debugCmd.AddCommand(stressCmd)
+	debugCmd.AddCommand(ioCmd)
+
+	debugCmd.PersistentFlags().StringVarP(&o.namespace, "namespace", "n", "default", "namespace to find chaos")
+	err = debugCmd.RegisterFlagCompletionFunc("namespace", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		return listNamespace(toComplete, c.KubeCli)
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	rootCmd.AddCommand(debugCmd)
+}
+
+// Run debug
+func (o *debugOptions) Run(chaosType string, args []string, c *cm.ClientSet) error {
+	if len(args) > 1 {
+		return fmt.Errorf("Only one chaos could be specified")
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	chaosName := ""
+	if len(args) == 1 {
+		chaosName = args[0]
+	}
+
+	chaosList, chaosNameList, err := cm.GetChaosList(ctx, chaosType, chaosName, o.namespace, c.CtrlCli)
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+	var result []cm.ChaosResult
+
+	for i, chaos := range chaosList {
+		var chaosResult cm.ChaosResult
+		chaosResult.Name = chaosNameList[i]
+
+		var err error
+		switch chaosType {
+		case "networkchaos":
+			err = networkchaos.Debug(ctx, chaos, c, &chaosResult)
+		case "stresschaos":
+			err = stresschaos.Debug(ctx, chaos, c, &chaosResult)
+		case "iochaos":
```

Comment:
Thanks for the suggestions! First time to know "magic values" 🙈  

Don't if it's ideal but I just predeclared them at the front, like `const networkChaos = "networkchaos"`. It's a bit redundant to use `iota` here, so it's not literally enum. Would that be acceptable?

Reply:
Well using Enums instead of directly using string is just to avoid dumb errors like mistyping. 😃

So when you use this enum, both string or `itoa` is acceptable for me. And you should expose the Enums for other packages could use it (if needed).

Reply:
Got it. Thanks for the explanation!!